test/libp2p/crypto/secp256k1_test.cpp

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <algorithm>
+
+#include <gtest/gtest.h>
+#include "testutil/outcome.hpp"
+#include "libp2p/crypto/secp256k1_provider/secp256k1_provider_impl.hpp"
+
+using libp2p::crypto::secp256k1::PrivateKey;
+using libp2p::crypto::secp256k1::PublicKey;
+using libp2p::crypto::secp256k1::Secp256k1ProviderImpl;
+using libp2p::crypto::secp256k1::Signature;
+
+/**
+ * @brief Pre-generated key pair and signature for sample message
+ * @using reference implementation from github.com/libp2p/go-libp2p-core
+ */
+
+#define SAMPLE_PRIVATE_KEY_BYTES                                             \
```

Comment:
you can use (if you want)
```
#include <libp2p/common/literals.hpp>
std::vector<uint8_t> vec = "D990F24FFC..."_unhex;
```

Reply:
BTW, pls avoid using `#define` if it not absolutely necessary, e.g. in source guards or when you write a macro function (actually better to avoid as well).
However, it may be ok in test cpp.

Reply:
We can't use `_unhex` here because type of private/public key is std::array and we don't need to create these unnecessary copy operations